scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -1370,6 +1406,544 @@ cdef class NumericalInversePolynomial(Method):
+    quintic Hermite interpolation to approximate the PPF by passing the `order`
+    parameter:
+
+    >>> class StandardNormal:
+    ...     def pdf(self, x):
+    ...        return 1/np.sqrt(2*np.pi) * np.exp(-x**2 / 2)
+    ...     def dpdf(self, x):
+    ...        return -1/np.sqrt(2*np.pi) * x * np.exp(-x**2 / 2)
+    ...     def cdf(self, x):
+    ...        return ndtr(x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> urng = np.random.default_rng()
+    >>> rng = NumericalInverseHermite(dist, order=5, random_state=urng)
+
+    Higher orders result in a fewer number of intervals:
+
+    >>> rng3 = NumericalInverseHermite(dist, order=3)
+    >>> rng5 = NumericalInverseHermite(dist, order=5)
+    >>> rng3.intervals, rng5.intervals
+    (3000, 522)
+
+    The u-error can be estimated by calling the `u_error` method. It runs a small
+    Monte-Carlo simulation to estimate the u-error. By default, 100,000 samples are
+    used. This can be changed by passing the `sample_size` argument:
+
+    >>> rng1 = NumericalInverseHermite(dist, u_resolution=1e-10)
+    >>> rng1.u_error(sample_size=1000000)  # uses one million samples
+    UError(max_error=9.53167544892608e-11, mean_absolute_error=2.2450136432146864e-11)
+
+    This returns a namedtuple which contains the maximum u-error and the mean
+    absolute u-error.
+
+    The u-error can be reduced by decreasing the u-resolution (maximum allowed u-error):
+
+    >>> rng2 = NumericalInverseHermite(dist, u_resolution=1e-13)
+    >>> rng2.u_error(sample_size=1000000)
+    UError(max_error=9.32027892364129e-14, mean_absolute_error=1.5194172675685075e-14)
+
+    Note that this comes at the cost of increased setup time and number of intervals.
+
+    >>> rng1.intervals
+    1022
+    >>> rng2.intervals
+    5687
+    >>> from timeit import timeit
+    >>> f = lambda: NumericalInverseHermite(dist, u_resolution=1e-10)
+    >>> timeit(f, number=1)
+    0.017409582000254886  # may vary
+    >>> f = lambda: NumericalInverseHermite(dist, u_resolution=1e-13)
+    >>> timeit(f, number=1)
+    0.08671202100003939  # may vary
+
+    Since the PPF of the normal distribution is available as a special function, we
+    can also check the x-error, i.e. the difference between the approximated PPF and
+    exact PPF::
+
+    >>> import matplotlib.pyplot as plt
+    >>> u = np.linspace(0.01, 0.99, 1000)
+    >>> approxppf = rng.ppf(u)
+    >>> exactppf = norm.ppf(u)
+    >>> error = np.abs(exactppf - approxppf)
+    >>> plt.plot(u, error)
+    >>> plt.xlabel('u')
+    >>> plt.ylabel('error')
+    >>> plt.title('Error between exact and approximated PPF (x-error)')
+    >>> plt.show()
+
+    """
+    cdef double[::1] construction_points_array
+
+    def __cinit__(self,
+                  dist,
+                  *,
+                  domain=None,
+                  order=3,
+                  u_resolution=1e-12,
+                  tol=None,
```

Comment:
Btw: What is the default u_res in PINV? Is it aligned? Probably 1e-10 would be sufficient.

Reply:
UNU.RAN has default `u_resolution` set to `1e-10`. I had to change this because previously `NumericalInverseHermite` had 1e-12 default `u_resolution`. But I agree that `1e-12` is too small and `1e-10` would be sufficient. I think we can change this without having major backward compatibility issues. If the setup worked with `1e-12` then it should also work with `1e-10` (and would be faster).

Reply:
Changed the default to `1e-10`